### PR TITLE
Add easystroke to available packages

### PR DESCRIPTION
**Pull request by aki-me**
###### Motivation for this change
Package not yet in list of available packages

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/easystroke/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+with pkgs;
+
+stdenv.mkDerivation rec {
+  name = "easystroke";
+  version = "patched-2018-12-05";
+  src = fetchTarball {
+    url = "https://github.com/thjaeger/easystroke/archive/refs/tags/0.6.0.tar.gz";
+    sha256 = "0kgzmwxqgbq3cgp2rvj09j9rk6h7ynbibim4mins3wzalgwy68dv";
+  };
```
Comment (lourkeur) [inline pkgs/applications/misc/easystroke/default.nix:14]:
```suggestion
  src = fetchFromGitHub {
    owner = "thjaeger";
    repo = "easystroke";
    rev = "0.6.0";
    sha256 = "0kgzmwxqgbq3cgp2rvj09j9rk6h7ynbibim4mins3wzalgwy68dv";
  };
```
Per guidelines, the most specific function should be used. Also note that `fetchTarball` is not meant to be used to fetch sources anyway.

File: pkgs/applications/misc/easystroke/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  nativeBuildInputs = [ gnutar bzip2 gnumake boost.dev gtkmm3.dev dbus-glib.dev pkg-config git xorg.xorgserver.dev ];
+  buildInputs = [ boost gtkmm3 xorg.libXtst intltool dbus-glib xorg.xorgserver ];
+  buildPhase = ''
+    # cd $src/easystroke
+    make
+  '';
```
Comment (lourkeur) [inline pkgs/applications/misc/easystroke/default.nix:20]:
```suggestion
```
based on my testing this is redundant.

File: pkgs/applications/misc/easystroke/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# based on easystroke-git from Arch AUR
+# as found on https://github.com/teu5us/easystroke-nix
+
+{ pkgs, ... }:
+
+with pkgs;
+
+stdenv.mkDerivation rec {
+  name = "easystroke";
+  version = "patched-2018-12-05";
```
Comment (lourkeur) [inline pkgs/applications/misc/easystroke/default.nix:10]:
```suggestion
  version = "0.6.0";
```
packages should be as close as possible to upstream vanilla.

File: pkgs/applications/misc/easystroke/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# based on easystroke-git from Arch AUR
+# as found on https://github.com/teu5us/easystroke-nix
+
+{ pkgs, ... }:
+
+with pkgs;
```
Comment (lourkeur) [inline pkgs/applications/misc/easystroke/default.nix:6]: using `{ stdenv, gnutar, <other dependencies> }:` is preferred since it allows overriding.

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]
Hi! Thank you for your contribution. I will use the [Contributing guidelines](https://nixos.org/manual/nixpkgs/stable/#reviewing-contributions-new-packages) to help you bring it up to Nixpkgs code standards.

In particular, some of these patches seem unnecessary. It's also not clear at all who wrote them and why, and they don't seem (?) to have been contributed upstream. If they have been, you should get them using `fetchpatch`. If they haven't, you should give a quick explanation of why they are needed using a comment in the nix file.

Your commit message should read "easystroke: init at 0.6.0"

This derivation also lacks metadata, in particular, description, license, and maintainer are required. Please refer to [this bit of the manual](https://nixos.org/manual/nixpkgs/stable/#chap-meta). Please also add yourself as a maintainer using a separate commit.

File: pkgs/applications/misc/easystroke/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/easystroke/default.nix:6]: actually using with pkgs and pkgs as inputs is in packages not allowed.

File: pkgs/applications/misc/easystroke/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/easystroke/default.nix:14]: Please subsitute the version.

File: pkgs/applications/misc/easystroke/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+stdenv.mkDerivation rec {
+  name = "easystroke";
+  version = "patched-2018-12-05";
+  src = fetchTarball {
+    url = "https://github.com/thjaeger/easystroke/archive/refs/tags/0.6.0.tar.gz";
+    sha256 = "0kgzmwxqgbq3cgp2rvj09j9rk6h7ynbibim4mins3wzalgwy68dv";
+  };
+  nativeBuildInputs = [ gnutar bzip2 gnumake boost.dev gtkmm3.dev dbus-glib.dev pkg-config git xorg.xorgserver.dev ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/easystroke/default.nix:15]:
```suggestion
  nativeBuildInputs = [ boost gtkmm3 dbus-glib.dev pkg-config git xorg.xorgserver ];
```
Why do we need git?
Most removed are in stdenv.
If any of those require only the headers they need to move to buildInputs.

File: pkgs/applications/misc/easystroke/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  patches = [
+    ./sigc.patch
+    ./dont-ignore-xshape-when-saving.patch
+    ./add-toggle-option.patch
+    ./abs.patch
+  ];
+  installPhase = ''
+    mkdir -p $out
+    make install PREFIX="$out"
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/easystroke/default.nix:30]: Missing meta.

Comment (SuperSandro2000) [inline pkgs/applications/misc/easystroke/default.nix:30]: Use makeFlags instead.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
From where did you get the patches? Can we fetch them from there?

**Pull request closed without merging**